server/server_test.go

```diff
@@ -1320,3 +1320,65 @@ func TestPropagatedShrinkExpandISR(t *testing.T) {
 	// Wait for ISR to expand.
 	waitForISR(t, 10*time.Second, name, 0, 2, s1, s2)
 }
+
+// Test stream pausing and resuming. A paused stream should re-activate it self
+// when a message is published using the Liftbridge API.
+func TestStreamPausing(t *testing.T) {
+	defer cleanupStorage(t)
+
+	// Use a central NATS server.
+	ns := natsdTest.RunDefaultServer()
+	defer ns.Shutdown()
+
+	// Configure server.
+	s1Config := getTestConfig("a", true, 5050)
+	s1 := runServerWithConfig(t, s1Config)
+	defer s1.Stop()
+
+	// Wait for server to elect itself leader.
+	getMetadataLeader(t, 10*time.Second, s1)
+
+	client, err := lift.Connect([]string{"localhost:5050"})
+	require.NoError(t, err)
+	defer client.Close()
+
+	// Create stream.
+	name := "foo"
+	subject := "foo"
+	err = client.CreateStream(context.Background(), subject, name)
+	require.NoError(t, err)
+
+	err = client.PauseStream(context.Background(), name)
+	require.NoError(t, err)
+
+	// Publish some messages.
+	num := 1
+	for i := 0; i < num; i++ {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+		_, err = client.Publish(ctx, name, []byte("hello"))
+		require.NoError(t, err)
+	}
+
+	// Force log clean.
+	forceLogClean(t, subject, name, s1)
+
+	// The first message read back should be the creation of the activity stream
+	// partition.
```

Comment:
Was the copy/pasted from the activity stream PR by mistake?

Reply:
Ah yes, I was a bit too hasty here. Should be fixed in 7025d22.